cookiecutter/zipfile.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import
+
+import os
+import requests
+import sys
+from zipfile import ZipFile
+
+from .prompt import read_user_yes_no
+from .utils import make_sure_path_exists, rmtree
+
+
+def prompt_and_delete(path, no_input=False):
+    """Ask the user whether it's okay to delete the previously-downloaded
+    file/directory.
+
+    If yes, deletes it. Otherwise, Cookiecutter exits.
+
+    :param path: Previously downloaded zipfile.
+    :param no_input: Suppress prompt to delete repo and just delete it.
+    """
+    # Suppress prompt if called via API
+    if no_input:
+        ok_to_delete = True
+    else:
+        question = (
+            "You've downloaded {} before. "
+            "Is it okay to delete and re-download it?"
+        ).format(path)
+
+        ok_to_delete = read_user_yes_no(question, 'yes')
+
+    if ok_to_delete:
+        if os.path.isdir(path):
+            rmtree(path)
+        else:
+            os.remove(path)
+    else:
+        sys.exit()
+
+    return ok_to_delete
+
+
+def unzip(zip_url, is_url, clone_to_dir='.', no_input=False):
+    # Ensure that clone_to_dir exists
+    clone_to_dir = os.path.expanduser(clone_to_dir)
+    make_sure_path_exists(clone_to_dir)
+
+    if is_url:
+        # Build the name of the cached zipfile,
+        # and prompt to delete if it already exists.
+        identifier = zip_url.rsplit('/', 1)[1]
+        zip_path = os.path.join(clone_to_dir, identifier)
```

Comment:
I'm voting for (3), largely because the mechanics of downloading and unzipping are just a means to an end (the template).

Also, I like my victory unpacked 😉